### PR TITLE
[WIP]Pytest Style analysis/test_density.py

### PR DESCRIPTION
**Pull request by utkbansal**
Fixes #

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by kain88-de** [comment]
build fails

**Comment by kain88-de** [comment]
build still fails

**Comment by utkbansal** [comment]
@kain88-de Can't figure what I broke in
```python
def test_grid(self, D):
        dV = D.delta.prod()  # orthorhombic grids only!
        # counts = (rho[0] * dV[0] + rho[1] * dV[1] ...) = sum_i rho[i] * dV
        assert_almost_equal(D.grid.sum() * dV, self.counts)
```

**Comment by utkbansal** [comment]
@kain88-de Need help here.

**Comment by kain88-de** [comment]
@jbarnoud or @mnmelo could you have a look?

**Comment by jbarnoud** [comment]
On [line 45](https://github.com/utkbansal/mdanalysis/blob/730396c5ec0eca162ca5c9342e568a6687d3d56e/testsuite/MDAnalysisTests/analysis/test_density.py#L45), as you declare class attributes, you use the class attribute `Lmax` to build the other class attribute `bins`. When you do this, it seems that python tries to find `Lmax` in the parent scope rather than using the `TestDensity.Lmax` you just defined.

I think you need to create a fixture for `bins`. This fixture can have the class as scope.

**Comment by kain88-de** [comment]
Yes python doesn't allow using the attributes you declare in a list comprehension

File: testsuite/MDAnalysisTests/analysis/test_density.py [hunk 1]
```diff
@@ -21,84 +21,90 @@
...
-from numpy.testing import (TestCase, assert_equal, assert_almost_equal, dec,
-                           assert_raises)
+from numpy.testing import assert_equal, assert_almost_equal
 
 import MDAnalysis as mda
+from MDAnalysis.analysis import density
 
 from MDAnalysisTests.datafiles import TPR, XTC, GRO
-from MDAnalysisTests import module_not_found, tempdir
+from MDAnalysisTests import tempdir
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_density.py]: please also remove the tempdir usage if you haven't done so already.

**Pull request merged**